distutils/tests/test_build.py

```diff
@@ -24,6 +25,8 @@ def test_finalize_options(self):
         # examples:
         #   build/lib.macosx-10.3-i386-cpython39
         plat_spec = f'.{cmd.plat_name}-{sys.implementation.cache_tag}'
+        if get_config_var('Py_GIL_DISABLED'):
+            plat_spec += 't'
```

Comment:
I confirmed that by adding this test (without the fix), the tests now fail, and tests pass with the fix. Nice.